src/index.ts

```diff
@@ -117,6 +117,29 @@ let resolver: Resolver | undefined
 const digestHashObject = (value: object | null | undefined) =>
   hashObject(value ?? {}).digest('hex')
 
+let bunCoreModules: Set<string> | undefined
+const getBunCoreModules = (): Set<string> => {
```

Comment:
It really seems hacking, can we ask @Jarred-Sumner for helping here.

Secondly, we can add a new module similar like https://www.npmjs.com/package/is-core-module.